src/daemon/link_connection.rs

```diff
@@ -28,62 +30,79 @@ use smolscale::{
 use sosistab2::{Multiplex, MuxSecret, Pipe};
 
 use super::{
-    context::{GLOBAL_IDENTITY, NEIGH_TABLE, RELAY_GRAPH},
+    context::{DEBTS, GLOBAL_IDENTITY, NEIGH_TABLE, RELAY_GRAPH},
     link_protocol::{AuthResponse, InfoResponse, LinkClient, LinkProtocol, LinkService},
     DaemonContext,
 };
 
 /// Encapsulates a single node-to-node connection (may be relay-relay or client-relay).
 #[derive(Clone)]
 pub struct LinkConnection {
-    mplex: Arc<Multiplex>,
     send_outgoing: Sender<RawPacket>,
     recv_incoming: Receiver<RawPacket>,
     remote_idpk: IdentityPublic,
-    _task: Arc<Immortal>,
+}
+
+pub struct LinkInfo {
+    pub conn: LinkConnection,
+    pub client: LinkClient,
+    pub task: Task<Infallible>,
 }
 
 impl LinkConnection {
     /// Creates a new Connection, from a single Pipe. Unlike in Geph, n2n Multiplexes in earendil all contain one pipe each.
-    pub async fn connect(ctx: DaemonContext, pipe: impl Pipe) -> anyhow::Result<Self> {
-        // First, we construct the Multiplex.
+    pub async fn connect(
+        ctx: DaemonContext,
+        pipe: impl Pipe,
+        max_outgoing_price: u64,
+    ) -> anyhow::Result<LinkInfo> {
```

Comment:
this should just return the LinkConnection. it's also overall a bad idea to have a function belong to a type, when it's not a method on its instance or a constructor.